ts/user-picker/user-picker.controller.ts

```diff
@@ -224,9 +224,35 @@ module lui.userpicker {
 			});
 		}
 
-		private refresh(clue: string = ""): ng.IPromise<IUserLookup[]> {
+		private refresh(clue: string = ""): ng.IPromise<any> {
+			return this.getUsers(clue).then(users => this.tidyUpAndAssign(users, clue));
+		}
+
+		// gets the users according to clue, also adds me and all if needed
+		// handles paging and customfilter
+		private getUsers(clue: string = ""): ng.IPromise<IUserLookup[]> {
+			let paging = this.$scope.lastPagingOffset + MAGIC_PAGING;
+			// only use paging if no customfilter
+			let cntTofetch = paging;
+			if (!!this.$scope.customFilter) {
+				cntTofetch = MAX_SEARCH_LIMIT;
+			}
+
+			let get = () => {
+				return this.userPickerService.getUsers(this.getFilter(clue), cntTofetch)
+				.then(users => {
+					if (!!this.$scope.customFilter) {
+						return _.chain(users)
+						.filter(u => this.$scope.customFilter(u))
```

Comment:
sauf erreur, tu devrais pouvoir écrire directement `.filter(this.$scope.customFilter)`, un poil plus joli